website/docs/reference/node-selection/state-comparison-caveats.md

```diff
@@ -16,7 +16,18 @@ dbt will mark modified any resource that depends on a changed macro, or on a mac
 
 ### Vars
 
-If a model uses a `var` or `env_var` in its definition, dbt is unable today to identify that lineage in such a way that it can include the model in `state:modified` because the `var` or `env_var` value has changed. It's likely that the model will be marked modified if the change in variable results in a different configuration.
+<VersionBlock lastVersion="1.8">
+
+If a model uses a `var` or `env_var` in its definition, dbt Core 1.8 and earlier are unable today to identify that lineage in such a way that it can include the model in `state:modified` because the `var` or `env_var` value has changed. It's likely that the model will be marked modified if the change in variable results in a different configuration.
+</VersionBlock>
+
+<VersionBlock firstVersion="1.9">
+
+Beginning in dbt Core 1.9, when you set the `state_modified_compare_vars` behavior flag to `True` and a model uses a `var` or `env_var` in its definition, dbt will identify that lineage in such a way that it can include the model in `state:modified` because the `var` or `env_var` value has changed.
```

Comment:
Do we also want to link to the new [state:modified.vars selection method](https://github.com/dbt-labs/docs.getdbt.com/pull/6222)?

Reply:
Yea not sure what I was thinking 😆 
moved here: https://github.com/dbt-labs/docs.getdbt.com/pull/6207/commits/ed01932e23cc92dc4f0abf31e8a519c5e972ffa6

Reply:
Linked. here https://github.com/dbt-labs/docs.getdbt.com/pull/6207/commits/84003bc8c576fdb404d19dbd9924d6fab1ed35d7